// artyom.js/source/examples.js
"use strict";
exports.__esModule = true;
var artyom_1 = require("./artyom");
// Create artyom instance
var Jarvis = new artyom_1["default"]();
// Add some commands in batch
Jarvis.addCommands([
    {
        indexes: ["Hello", "Hi"],
        action: function () {
            Jarvis.say("Hello");
        }
    }
]);
// Add a single command with less lines
Jarvis.on(["Good Morning"]).then(function () {
    Jarvis.say("Good Morning !");
});
// Initialize and Start Artyom
Jarvis.initialize({
    debug: true,
    continuous: true,
    lang: "en-GB",
    listen: true
}).then(function () {
    console.log("Jarvis succesfully initialized");
})["catch"](function () {
    console.log("Oops, something went wrong with your configuration ... ");
});
// Stop Artyom
setTimeout(function () {
    Jarvis.fatality().then(function () {
        console.log("Jarvis succesfully stopped !");
    })["catch"](function () {
        console.log("Well, this shouldn't happen :) ... ");
    });
}, 5000);
// Restart Artyom
Jarvis.restart().then(function () {
    console.log("Artyom succesfully restarted with the same initialization configuration");
})["catch"](function (err) {
    console.error("Cannot restart", err);
});
